modules/broker-core/src/main/java/org/wso2/broker/core/QueueHandler.java

```diff
@@ -217,4 +221,15 @@ public int consumerCount() {
         return consumers.size();
     }
 
+    public void addBinding(Binding binding, ThrowingConsumer<Binding, BrokerException> bindingChangeListener) {
```

Comment:
Can we just pass only the listener and maintain a list? We can keep the corresponding binding inside the listener.

Reply:
Used a map 
1. To get all the bindings directly from the queue for REST API (Not implemented yet)
1. To avoid using a new listener object per each binding. Currently we have a listener per each binding registry/Exchange.

Reply:
👍